internal/bsr/bsr.go

```diff
@@ -237,6 +238,173 @@ func OpenSession(ctx context.Context, sessionRecordingId string, f storage.FS, k
+
+	validation.ChecksumValidation = sessionChecksumValidation
+
+	conns := session.Meta.connections
+
+	for connId := range conns {
+		// Get connection id
+		connKey := connId
+		lastDotIndex := strings.LastIndex(connId, ".")
+		if lastDotIndex != -1 {
+			connKey = connId[:lastDotIndex]
+		}
+
+		connectionValidation := Validation{
+			Name:         connId,
+			Type:         ConnectionContainer,
+			SubContainer: []Validation{},
+		}
+
+		conn, err := session.OpenConnection(ctx, connKey)
+		if err != nil {
+			validationSummary.Valid = false
+			connectionValidation.Error = errors.Join(connectionValidation.Error, fmt.Errorf("%s: failed to retrieve connection for %s: %w", op, connId, err))
+			continue
+		}
+
+		// Validate current connection
+		connectionChecksumValidation, err := conn.container.ValidateChecksums(ctx)
+		if err != nil {
+			validationSummary.Valid = false
+			connectionValidation.Error = errors.Join(connectionValidation.Error, fmt.Errorf("%s: failed to validate connection for %s: %w", op, connId, err))
+			continue
+		}
+
+		validationSummary = updateValidationSummary(ctx, ConnectionContainer, connId, connectionChecksumValidation, validationSummary)
+
+		connectionValidation.ChecksumValidation = connectionChecksumValidation
+
+		chs := conn.Meta.channels
+
+		for chId := range chs {
+			// Get channel id
+			chsKey := chId
+			lastDotIndex := strings.LastIndex(chId, ".")
+			if lastDotIndex != -1 {
+				chsKey = chId[:lastDotIndex]
+			}
+
+			channelValidation := Validation{
+				Name:         chId,
+				Type:         ChannelContainer,
+				SubContainer: []Validation{},
+			}
+
+			ch, err := conn.OpenChannel(ctx, chsKey)
+			if err != nil {
+				validationSummary.Valid = false
+				channelValidation.Error = errors.Join(channelValidation.Error, fmt.Errorf("%s: failed to retrieve channel for %s: %w", op, chId, err))
+				continue
+			}
+
+			// Validate current channel
+			channelChecksumValidation, err := ch.container.ValidateChecksums(ctx)
+			if err != nil {
+				validationSummary.Valid = false
+				channelValidation.Error = errors.Join(channelValidation.Error, fmt.Errorf("%s: failed to validate channel for %s: %w", op, chId, err))
+				continue
+			}
+
+			validationSummary = updateValidationSummary(ctx, ChannelContainer, chId, channelChecksumValidation, validationSummary)
+			channelValidation.ChecksumValidation = channelChecksumValidation
+
+			connectionValidation.SubContainer = append(connectionValidation.SubContainer, channelValidation)
+		}
+
+		validation.SubContainer = append(validation.SubContainer, connectionValidation)
+	}
+
+	return validation, validationSummary, nil
+}
+
+// updateValidationSummary updates the ValidationSummary based on the failed checksums.
+// The updated ValidationSummary is then returned
+func updateValidationSummary(ctx context.Context, c ContainerType, id string, cv ChecksumValidation, vs ValidationSummary) ValidationSummary {
```

Comment:
What if this was a pointer receiver that updated the underlying ValidationSummary directly?